dev-lang/fpc/fpc-3.0.2-r1.ebuild

```diff
@@ -0,0 +1,166 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=5
+
+inherit toolchain-funcs
+
+HOMEPAGE="https://www.freepascal.org/"
+DESCRIPTION="Free Pascal Compiler"
+SRC_URI="mirror://sourceforge/freepascal/fpcbuild-${PV}.tar.gz
+	mirror://sourceforge/freepascal/fpc-${PV}.source.tar.gz
+	amd64? ( mirror://sourceforge/freepascal/${P}.x86_64-linux.tar )
+	x86? ( mirror://sourceforge/freepascal/${P}.i386-linux.tar )
+	doc? ( mirror://sourceforge/freepascal/Documentation/${PV}/doc-html.tar.gz -> ${P}-doc-html.tar.gz )"
+
+SLOT="0"
+LICENSE="GPL-2 LGPL-2.1-with-linking-exception"
+KEYWORDS="~amd64 ~arm ~ppc ~x86"
+IUSE="doc ide source"
+
+RDEPEND="ide? ( !dev-lang/fpc-ide )"
+
+RESTRICT="strip" #269221
+
+S=${WORKDIR}/fpcbuild-${PV}/fpcsrc
+
+pkg_pretend() {
+	if [[ ${MERGE_TYPE} != binary ]]; then
+		# Bug 475210
+		if $(tc-getLD) --version | grep -q "GNU gold"; then
+			eerror "fpc does not function correctly when built with the gold linker."
+			eerror "Please select the bfd linker with binutils-config."
+			die "GNU gold detected"
+		fi
+	fi
+}
+
+src_unpack() {
+	case ${ARCH} in
+		amd64)	FPC_ARCH="x86_64"    PV_BIN=${PV} ;;
+		x86)	FPC_ARCH="i386"      PV_BIN=${PV} ;;
+		*)	die "This ebuild doesn't support ${ARCH}." ;;
+	esac
+
+	unpack ${A}
+
+	tar -xf ${PN}-${PV_BIN}.${FPC_ARCH}-linux/binary.${FPC_ARCH}-linux.tar || die "Unpacking binary.${FPC_ARCH}-linux.tar failed!"
+	tar -xzf base.${FPC_ARCH}-linux.tar.gz || die "Unpacking base.${FPC_ARCH}-linux.tar.gz failed!"
+}
+
+src_prepare() {
+	find "${WORKDIR}" -name Makefile -exec sed -i -e 's/ -Xs / /g' {} + || die
+
+	# let the pkg manager compress man files
+	sed -i '/find man.* gzip /d' "${WORKDIR}"/fpcbuild-${PV}/install/man/Makefile || die
+
+	# make the compiled binary check for fpc.cfg under the prefixed /etc/ path
+	sed -i "s|if CfgFileExists('/etc/'+fn) then|if CfgFileExists('${EPREFIX}/etc/'+fn) then|g" "${WORKDIR}"/fpcbuild-${PV}/fpcsrc/compiler/options.pas || die
+	sed -i "s|foundfn:='/etc/'+fn|foundfn:='${EPREFIX}/etc/'+fn|g" "${WORKDIR}"/fpcbuild-${PV}/fpcsrc/compiler/options.pas || die
+}
+
```

Comment:
This can be simplified by hprefixify from prefix.eclass, to `hprefixify "${WORKDIR}"/fpcbuild-${PV}/fpcsrc/compiler/options.pas`.